packages/aws-cdk/lib/api/aws-auth/credential-plugins.ts

```diff
@@ -65,13 +66,13 @@ export class CredentialPlugins {
       // Otherwise it must have returned credentials.
       const credentials = (providerOrCreds as any).resolvePromise
```

Comment:
The `as any` should no longer be necessary, should it?

I don't see `resolvePromise()` in the typing package. Where is it coming from? What is this object?